test/test-batch-transfer.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/sh
+# This is a sample Git LFS test.  See test/README.md and testhelpers.sh for
+# more documentation.
+
+. "test/testlib.sh"
+
+begin_test "batch transfer"
+(
+  set -e
+
+  # This initializes a new bare git repository in test/remote.
+  # These remote repositories are global to every test, so keep the names
+  # unique.
+  reponame="$(basename "$0" ".sh")"
+  setup_remote_repo "$reponame"
+
+  # Clone the repository from the test Git server.  This is empty, and will be
+  # used to test a "git pull" below. The repo is cloned to $TRASHDIR/clone
+  clone_repo "$reponame" clone
+
+  # Clone the repository again to $TRASHDIR/repo. This will be used to commit
+  # and push objects.
+  clone_repo "$reponame" repo
+
+  # This executes Git LFS from the local repo that was just cloned.
+  out=$($GITLFS track "*.dat" 2>&1)
+  echo "$out" | grep "Tracking \*.dat"
+
+  contents=$(printf "a")
+  contents_oid=$(printf "$contents" | shasum -a 256 | cut -f 1 -d " ")
+
+  printf "$contents" > a.dat
+  git add a.dat
+  git add .gitattributes
+  out=$(git commit -m "add a.dat" 2>&1)
+  echo "$out" | grep "master (root-commit)"
+  echo "$out" | grep "2 files changed"
+  echo "$out" | grep "create mode 100644 a.dat"
+  echo "$out" | grep "create mode 100644 .gitattributes"
```

Comment:
I've started using this pattern:

```shell
git commit -m "add a.dat" 2>&1 | tee commit.log
grep "master (root-commit)" commit.log
```

Nice thing is that the output is only printed to STDOUT once with `tee`.  If the last grep fails, you'll only see the commit output once, and the matches of the successful grep commands.